spec/factories/ext_management_system.rb

```diff
@@ -0,0 +1,39 @@
+FactoryBot.define do
+  factory :ems_oracle_cloud, :class => "ManageIQ::Providers::OracleCloud::CloudManager", :parent => :ems_cloud do
+    provider_region { "us-ashburn-1" }
+  end
+
+  factory :ems_oracle_cloud_with_vcr_authentication, :parent => :ems_oracle_cloud do
+    uid_ems { "ocid1.tenancy.oc1..aaaaaaaa" }
+
+    after(:create) do |ems|
+      # These are all bogus values but Rails Secrets won't work because the replaced
+      # values don't pass the local sdk client validation
+      user_id     = "ocid1.user.oc1..aaaaaaaa"
+      private_key = <<~PRIVATE_KEY
+        -----BEGIN RSA PRIVATE KEY-----
+        MIIBOwIBAAJBAMfW33yRX0zpGhFx8kvQcpip1pHG0QJVpUp7ik4W9JLl0PWqSudF
+        eXF0hFAc+Vx9R7ufpxMZ4lp+10WetdObAzsCAwEAAQJBAIw8R1y9DymDst1nHucB
+        AkoLdR2bbXS78WBRTX77MOobynsP5gJehUI8hEOkbzYIJNBlSS/3aFAHBrd72diL
+        ZPECIQDnr3y/pug4gNsoOeDWPGPxmLbd8cK9LS4KA0jeQ2PbfQIhANzPzNaHwZ5R
+        aV/TnK00HpNJQBSo4S4Qbfc6JkZsD2cXAiEAiMI4s/R07S16sBMCGdPJ9wl7ICWe
+        Gwb5PyXTNIe5AQ0CIC5mzJjYdmuamBY3Fdmf9jzlS74LryZK9ZDae2iZFLOJAiA6
+        7KHzcqovm0lETCaEKSMWhKEcIfmz4qbAqxbx+DmAlQ==
+        -----END RSA PRIVATE KEY-----
+      PRIVATE_KEY
+      public_key = <<~PUBLIC_KEY
+        -----BEGIN PUBLIC KEY-----
+        MFwwDQYJKoZIhvcNAQEBBQADSwAwSAJBAMfW33yRX0zpGhFx8kvQcpip1pHG0QJV
+        pUp7ik4W9JLl0PWqSudFeXF0hFAc+Vx9R7ufpxMZ4lp+10WetdObAzsCAwEAAQ==
+        -----END PUBLIC KEY-----
+      PUBLIC_KEY
```

Comment:
None of these work well as rails secrets because the `oci-ruby-sdk` fails on e.g. `ORACLE_CLOUD_USER_ID` having an invalid format for a user id and SSL library fails when you pass e.g. `ORACLE_CLOUD_PRIVATE_KEY`

Not sure if there is a better way to do this but it really only makes it more difficult for someone to re-record vcrs (replace these with the real values and add `config.filter_sensitive_data` for the user_id/tenant_id) which isn't a huge burden.

Reply:
weird...you'll have to show me the current Rails secrets stuff, because I'd like to understand that better even from just a pluggable providers consistency standpoint (we don't use it everywhere, but probably should)

Reply:
:+1: it builds the constant based on the path in the yaml file, so if I have:
```
config/secrets.yml
---
test:
  oracle_cloud:
    user_id: "abcd"
```

Then it will automatically replace `"abcd"` in the vcr cassettes with `ORACLE_CLOUD_USER_ID`.  

Then in our spec tests we do:
```
Rails.application.secrets.test&.oracle_cloud&.user_id || "ORACLE_CLOUD_USER_ID"
```

so that if you don't have the secrets file it'll match what is in the vcr cassette.

This works great as long as you can pass in `ORACLE_CLOUD_USER_ID` to the sdk client, but if that client does any format verification you can't control what it replaces the secret string with as far as I've been able to find.